CMakeLists.txt

```diff
@@ -109,9 +111,24 @@ list(APPEND CMAKE_PREFIX_PATH /opt/rocm /opt/rocm/llvm /opt/rocm/hip /opt/rocm/h
 option(ENABLE_HIP_WORKAROUNDS Off)
 set(MIOPEN_INSTALL_CXX_HEADERS Off CACHE BOOL "Install MIOpen's C++ header interface")
 
-option(MIOPEN_USE_COMGR "When possible, use comgr to build kernels instead of offline tools" Off)
+
+# Embedded Build Configuration
+option(MIOPEN_EMBED_BUILD "Build with the set of embed flags." Off)
+option(MIOPEN_USE_COMGR "Use comgr to build kernels instead of offline tools" ${MIOPEN_EMBED_BUILD})
+option(MIOPEN_DISABLE_USERDB "Disable user database access" ${MIOPEN_EMBED_BUILD})
 option(MIOPEN_USE_HIP_KERNELS "Use HIP kernels." On)
 
+# Workaround for comgr 
+if(MIOPEN_USE_COMGR)
+    set(MIOPEN_USE_HIP_KERNELS 0)
+endif()
+
+# Static build
+if(MIOPEN_EMBED_BUILD)
+    set(BUILD_SHARED_LIBS 0)
```

Comment:
We shouldn't shadow the cached variable. It can be confusing to users when they look at the cache. You could set the value like this:

```cmake
if(MIOPEN_EMBED_BUILD)
option( BUILD_SHARED_LIBS "Build as a shared library" Off )
else()
option( BUILD_SHARED_LIBS "Build as a shared library" On )
endif()
```

But it might be better to not set `BUILD_SHARED_LIBS` based on `MIOPEN_EMBED_BUILD` at all. Devops will still set `-DBUILD_SHARED_LIBS=Off` on all the builds anyways. And the customer that might use this is not going to be using the `BUILD_SHARED_LIBS` variable.

Reply:
The point of this particular PR is to have one flag that configure the embedded build. `BUILD_SHARED_LIBS=Off` is one of those requirements. Otherwise the user can configure with it off, or on as they wish.

Reply:
Fixed.